src/server/db/actions/UserAction.ts

```
@@ -51,4 +53,78 @@ export async function verifyUser(email: string, password: string) {
       message: "Invalid password",
     };
   }
+
+}
+
+export async function getUser(email: string) {
+  await connectMongoDB();
+  try {
+    const user = await UserModel.findOne({ email: email });
+    return user;
+  } catch (e) {
+    throw e;
+  }
+}
+
+export async function editUser(data: any) {
+  await connectMongoDB();
+  try {
+    const result = await UserModel.findByIdAndUpdate(data._id, data, {
+      new: true,
+    });
+    if (!result) {
+      throw new ReferenceError("User with given ID does not exist.");
+    }
+    const user = await UserModel.findById(data._id);
+  } catch (e) {
+    throw e;
+  }
 }
+
+export async function editPassword(data: any) {
+  await connectMongoDB();
+
+  try {
+    const user = await UserModel.findOne({ email: data.email });
+    if (!user) {
+      return {
+        status: 404,
+        message: "User does not exist",
+      };
```

We should throw an error here instead of returning an object with this information. The action should be independent of the controller that calls the action - the action shouldn't include information about the HTTP status code.